mpd/client.go

```diff
@@ -48,6 +55,46 @@ type Client struct {
 	text *textproto.Conn
 }
 
+// CommandError represents an error returned by the MPD server.
+// It contains the error number, the index of the causing command in the command list,
+// the name of the command in the command list and the error message.
+type CommandError struct {
+	Code             CommandErrorCode
+	CommandListIndex int
+	CommandName      string
+	Message          string
+}
+
+// CommandErrorCode is the error code of a CommandError.
+type CommandErrorCode int
+
+// CommandErrorCodes as defined in MPD source (src/protocol/Ack.hxx) version 0.21.7.
+const (
+	AckErrorNotList CommandErrorCode = iota + 1
```

Comment:
`Ack` is protocol detail that user may not know about. Name this ErrorCodeNotList? Similarly, for the other error codes.